gulpfile.js

```diff
@@ -1,29 +1,101 @@
+const { spawn } = require('child_process');
+
+const del = require('del');
 const forever = require('forever-monitor');
 const gulp = require('gulp');
+const gulpBabel = require('gulp-babel');
 const eslint = require('gulp-eslint');
 const mocha = require('gulp-mocha');
+const sourcemaps = require('gulp-sourcemaps');
+const webpackStream = require('webpack-stream');
+const webpackConfig = require('./webpack.config.js');
+
+/**
+ * @summary Delete the dist/ directory
+ */
+const babelClean = () => del(['dist']);
+
+/**
+ * @summary Copy files that don't need to be compiled from src/ to dist/
+ */
+const babelCopy = () => gulp.src(['src/**', '!src/**/*.js', '!src/tests/integration/**'])
+  .pipe(gulp.dest('dist'));
+
+/**
+ * @summary Compile JavaScript files and place them in dist/. Also, generate sourcemaps
+ */
+const babelCompile = () => gulp.src(['src/**/*.js'])
+  .pipe(sourcemaps.init())
+  .pipe(gulpBabel())
+  .pipe(gulp.dest('dist'));
+
+const babel = gulp.series(babelClean, babelCopy, babelCompile);
 
 /**
  * @summary Use Eslint linting *.js file besides source files in node_modules
  */
-gulp.task('lint', () => gulp.src(['**/*.js', '!node_modules/**'])
+const lint = () => gulp.src(['src/**/*.js', '!node_modules/**', '!dist/**'])
   .pipe(eslint())
   .pipe(eslint.format())
-  .pipe(eslint.failAfterError()));
+  .pipe(eslint.failAfterError());
+
+/**
+ * @summary Check Flow types
+ */
+const typecheck = () => spawn('./node_modules/.bin/flow', ['check'], { stdio: 'inherit' });
+
+/**
+ * @summary Run unit tests
+ */
+const test = () => gulp.src(['dist/tests/unit/*.js'])
+  .pipe(mocha({ reporter: 'spec' }));
 
 /**
- * @summary Run unit test
+ * @summary Bundle using webpack
  */
-gulp.task('test', () => gulp.src(['tests/unit/*.js'])
-  .pipe(mocha({ reporter: 'spec' })));
+const bundle = () => gulp.src('dist/app.js')
+  .pipe(webpackStream(webpackConfig))
+  .pipe(gulp.dest('dist/'));
```

Comment:
The bundle task requires that the babel task has been completed. I cannot think of why it would be useful to only bundle without running babel beforehand, since that will "compile" any changes made in `src/`. So, I will add the babel task in series with bundle